server.py

```diff
@@ -62,7 +64,32 @@ def run_tasks(services):
         logging.info('All systems ready.')
         loop.run_forever()
     except KeyboardInterrupt:
-        loop.run_until_complete(services.get('app_svc').teardown())
+        loop.run_until_complete(services.get('app_svc').teardown(main_config_file=args.environment))
+
+
+def make_secure_config(config_name):
```

Comment:
this function has multiple levels of abstraction in it and I'd recommend refactoring.

pull out the reading and writing of the config file into separate functions
separate out the log credentials code block

this should make this function easier to read.

Reply:
I've broken this up and stored the new functions in a new utility module https://github.com/mitre/caldera/pull/1544/files#diff-6449db3e452640506e59349e927485e8